js/Schemas/pluginDefinitionSchema.json

```diff
@@ -0,0 +1,345 @@
+{
+   "$schema": "http://json-schema.org/schema#",
+   "type": "object",
+   "if": {
+      "properties": {
+         "pluginType": {
+            "const": "library"
+         }
+      }
+   },
+   "then": {
+      "required": [
+         "libraryName",
+         "libraryVersion"
+      ]
+   },
+   "else": {
+      "if": {
+         "properties": {
+            "pluginType": {
+               "const": "nodeAuthentication"
+            }
+         }
+      },
+      "then": {
+         "required": [
+            "authenticationCategory",
+            "filename"
```

Comment:
"filename" is fine except that it breaks everything because we incorrectly were using "fileName". Can you fix configjs and terminalproxy within this same pull request? We'll just need to open up other pull requests to fix the other repos dependent upon this one.